sublime_jedi/helper.py

```diff
@@ -33,28 +48,97 @@ def docstring_panel(view, docstring):
         sublime.status_message('Jedi: No results!')
 
 
-def docstring_tooltip(view, docstring, content_builder):
+def docstring_tooltip(view, docstring, location=None):
     """Show docstring in popup.
 
     :param view (sublime.View): current active view
     :param docstring (basestring): python __doc__ string
-    :param content_builder (callable): callable object should accept docstring
-        and return content read for popup (a html text)
+    :param location (int): The text point where to create the popup
     """
-    if docstring:
-        content = content_builder(docstring)
-        view.show_popup(content, max_width=512)
+    if not docstring:
+        return sublime.status_message('Jedi: No results!')
+
+    if location is None:
+        location = view.sel()[0].begin()
+
+    # fallback if mdpopups is not available
+    if not _HAVE_MDPOPUPS:
```

Comment:
Basically there are 2 renderers only:

1. Using mdpopups to highlight the signature and rendering the rest of the docstring as markdown.
2. If mdpopups is not available the current one is used.
> I think next step will be: add a .rst renderer.

The markdown rendering fully depends on mdpopups package and its capabilities. Not sure whether it will support restructured text some day.

I already thought about highlighting at least the :param: parts, but this will go far beyound the helper module. 

> I think we should allow "setup" renderer in plugin options.

Why do you setup the renderer? It should detect the format automatically and render the best it can. mdpopups doesn't do any harm as it handles everything as normal text. The only question of taste might be syntax highlighting of the signature, I think.

> We should be less coupled with md and simple renderer.

You want the renderer function passed as argument again? I removed it for not in order not to bloat both `SublimeJediDocstring` and `SublimeJediTooltip` with the decision about which one to use. But your are right: This might not be the cleanest/best solution. On this stage it is more or less a case study and a proposal about what might be possible.